### PR TITLE
SALTO-7078: Override zip timestamps in static resources in SFDX flow

### PR DESCRIPTION
**Pull request by yelly**
Currently we get a diff when comparing two identical SFDX projects because the zip is built dynamically and the timestamps within zip files are different.

---

_Additional context for reviewer_

---

_Release Notes_: 
None.

---

_User Notifications_: 
None.

### CODE REVIEW

File: packages/salesforce-adapter/src/filters/static_resource_zip_timestamps.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    const zipStaticResources = elements
+      .filter(isStaticResourceInstance)
+      .filter(inst => inst.value.contentType === 'application/zip')
+    await awu(zipStaticResources).forEach(async inst => {
+      const staticFile = inst.value[METADATA_CONTENT_FIELD]
+      const content = await staticFile.getContent()
+      if (content === undefined) {
+        return
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/filters/static_resource_zip_timestamps.ts:55]: is this scenario ever expected? If not, let's add an error/warn log

File: packages/salesforce-adapter/test/filters/static_resource_zip_timestamps.test.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import JSZip from 'jszip'
+import { InstanceElement, StaticFile } from '@salto-io/adapter-api'
+import filterCreator, { StaticResourceInstance } from '../../src/filters/static_resource_zip_timestamps'
+import { METADATA_CONTENT_FIELD } from '../../src/constants'
+import { defaultFilterContext } from '../utils'
+import mockClient from '../client'
+import { FilterWith } from './mocks'
+import { mockTypes } from '../mock_elements'
+
+describe('Static Resource File Extension Filter', () => {
```
Comment (tamtamirr) [inline packages/salesforce-adapter/test/filters/static_resource_zip_timestamps.test.ts:17]: Missing UT for non SFDX flow (client is defined)

**Review by tamtamirr — COMMENTED** [review]
LGTM! Added a few comments

File: packages/salesforce-adapter/src/filters/static_resource_zip_timestamps.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  value: {
+    contentType: string
+    [METADATA_CONTENT_FIELD]: StaticFile
+  }
+}
+
+const replaceZipTimestamp = async (data: Buffer): Promise<Buffer> => {
+  const zip = await JSZip.loadAsync(data)
+  zip.forEach((_relativePath, file) => {
+    file.date = DATE
```
[the remark this answers is not in the document]
Comment (yelly) [reply packages/salesforce-adapter/src/filters/static_resource_zip_timestamps.ts:30]: Done.

File: packages/salesforce-adapter/src/filters/static_resource_zip_timestamps.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  name: 'staticResourceZipTimestampFilter',
+  onFetch: async (elements: Element[]): Promise<void> => {
+    // Run only in the SFDX flow.
+    if (client !== undefined) {
+      return
+    }
+
+    const zipStaticResources = elements
+      .filter(isStaticResourceInstance)
+      .filter(inst => inst.value.contentType === 'application/zip')
```
[the remark this answers is not in the document]
Comment (yelly) [reply packages/salesforce-adapter/src/filters/static_resource_zip_timestamps.ts:50]: Done.

File: packages/salesforce-adapter/src/filters/static_resource_zip_timestamps.ts (hunk 1 above)
Comment (yelly) [reply packages/salesforce-adapter/src/filters/static_resource_zip_timestamps.ts:55]: Done.

File: packages/salesforce-adapter/test/filters/static_resource_zip_timestamps.test.ts (hunk 2 above)
Comment (yelly) [reply packages/salesforce-adapter/test/filters/static_resource_zip_timestamps.test.ts:17]: It's the 2nd describe...

Comment (tamtamirr) [reply packages/salesforce-adapter/test/filters/static_resource_zip_timestamps.test.ts:17]: im blind sorry 🙏

**Review by tamtamirr — APPROVED** [review]

**Pull request merged**